endpoints/openrtb2/auction.go

```diff
@@ -283,6 +284,14 @@ func (deps *endpointDeps) validateRequest(req *openrtb.BidRequest) []error {
 		errL = append(errL, &errortypes.Warning{Message: fmt.Sprintf("A prebid request can only process one currency. Taking the first currency in the list, %s, as the active currency", req.Cur[0])})
 	}
 
+	// If automatically filling source TID is enabled then validate that
+	// source.TID exists and If it doesn't, fill it with a randomly generated UUID
+	if deps.cfg.AutoSourceTIDFill {
+		if err := validateAndFillSourceTID(req); err != nil {
+			return []error{err}
+		}
+	}
+
```

Comment:
Can we add a `validateRequest` test case to cover the true branch of the newly added if statement?

![image](https://user-images.githubusercontent.com/15863840/91228038-8c184c00-e6f5-11ea-972f-ea32b8c930f9.png)